src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -636,7 +636,21 @@ private function restoreResponseBody(Request $request, Response $response)
             if ($response->headers->has('X-Body-File')) {
                 include $response->headers->get('X-Body-File');
```

Comment:
There's no need to change this: the required file cannot come from any user input, and we do validate that the name of the file, so that arbitrary file inclusion is not possible.

Reply:
Well, from my understanding of re-reading this code, this `X-Body-File` corresponds to a file written in the cache store. If we stop using PHP to implement the ESI logic, we might need to process the boundaries there instead of evaluating the file.

Do we have functional tests covering the case of a processing of ESI tags alongside a cached response of HttpCache ? Because I suspect that this is the case that is now broken (we would write the cache with boundaries instead of PHP code but read it as if it was PHP code).